[AV2-1207] UX/UI Manual Save button and last save time in assessment UI

sasangachathumal opened the pull request:
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-1207

**Additional Comments:**
Updated last save section hight and save button hight.
update last save section padding.
update last save to update through events (rxjs subjects).
update activity-desktop, assessment-mobile to broadcast that event.

**Checklist:**

- [x] Unit test completed?: (Y/N)
- [x] Docs folder up to date?: (Y/N)
- [x] Add if anything missed: (Y/N)

All checklists are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

sasangachathumal committed:
[AV2-1207]
Updated last save section hight and save button hight.
update last save section padding.
update last save to update throgh events (rxjs subjects).
update activity-desktop, assessment-mobile to broadcast that event.

sasangachathumal commented:
Please check jira story comments for video how it works now.

shawnm0705 commented on projects/v3/src/app/components/assessment/assessment.component.ts:
```diff
@@ -95,6 +96,22 @@ export class AssessmentComponent implements OnChanges {
+   * this event will update saving message when saving in progress and done.
+   * this event is broadcasting from activity-desktop and assessment-mobile components.
+   */
+  private _initSavingEvent() {
```
This `assessment.component` is a stand-alone component, which shouldn't subscribe to a global wide event.
Why do we need to check the saving message outside of this component?

sasangachathumal committed:
[AV2-1207]
moved get current day to util service.
added new input to assessment component to manage save text.
create method to populoate last save text.
added code to pass new input to assessment component in every ware use.

sasangachathumal committed:
[AV2-1207] removed return becouse it effect review

shawnm0705 commented on projects/v3/src/app/components/assessment/assessment.component.ts:
```diff
@@ -90,11 +91,28 @@ export class AssessmentComponent implements OnChanges {
+   * after save done parent component change last save to current time.
+   */
+  private _populateLastSaveText() {
+    if (!this.savingMessage || this.savingMessage === '') {
```
instead of this, we can simply put
```
if (this.savingMessage) {
return;
}
```

shawnm0705 commented on projects/v3/src/app/pages/assessment-mobile/assessment-mobile.page.ts:
```diff
@@ -85,12 +88,16 @@ export class AssessmentMobilePage implements OnInit {
       this.notificationsService.assessmentSubmittedToast();
       // get the latest activity tasks and refresh the assessment submission data
       this.activityService.getActivity(this.activityId);
+      this.savingText = 'Last saved ' + this.utils.getFormatedCurrentTime();
       return this.assessmentService.getAssessment(this.assessment.id, this.action, this.activityId, this.contextId, this.submissionId);
+    } else {
+      this.savingText = 'Last saved ' + this.utils.getFormatedCurrentTime();
```
the savingText is changed for both `if` and `else`, why don't we just bring it out of the `if,else` then?

sasangachathumal committed:
[AV2-1207]
did some changes mention in code review.

shawnm0705 commented on projects/v3/src/app/pages/activity-desktop/activity-desktop.page.ts:
```diff
@@ -76,6 +78,7 @@ export class ActivityDesktopPage implements OnInit {
     } else {
       this.loading = false;
     }
+    this.savingText = 'Last saved ' + this.utils.getFormatedCurrentTime();
```
the function will return `if (!event.assessment.inProgress)` so this should come before that

sasangachathumal committed:
[AV2-1207]
fixed code review things
